core/buidler.config.js

```diff
@@ -0,0 +1,42 @@
+const chalkPipe = require("chalk-pipe");
+const { usePlugin, task } = require("@nomiclabs/buidler/config");
+
+usePlugin("@nomiclabs/buidler-truffle5");
+usePlugin("solidity-coverage");
+
+task("test")
+  .addFlag("debug", "Compile without optimizer")
+  .setAction(async (taskArgs, bre, runSuper) => {
+    const { debug } = taskArgs;
+
+    if (debug) {
+      bre.config.solc.optimizer.enabled = false;
+      console.log(chalkPipe("bold.underline")("Running tests in debug mode"));
+    }
+
+    await runSuper(taskArgs);
+  });
+
+module.exports = {
+  solc: {
+    version: "0.6.6",
+    optimizer: {
+      enabled: true,
+      runs: 200
+    }
+  },
+  networks: {
+    buidlerevm: {
+      gas: 12000000,
+      blockGasLimit: 0x1fffffffffffff,
+      allowUnlimitedContractSize: true,
```

Comment:
One last nit, thoughts on setting these to:
```
      gas: 11500000,
      blockGasLimit: 11500000,
      allowUnlimitedContractSize: false,
```
to better mirror production?

The gas and gas limit probably don't matter _that_ much since it's unlikely we would _accidentally_ create a transaction that big unless it were a deployment that exceeded the max contract size anyway. However, the block gas limits are currently hovering [a little under 12MM](https://etherscan.io/chart/gaslimit), so I feel like it might make sense to just set them both to 11.5MM just to be safe. The max contract size _is_ something we've hit in the past, so I'd suggest that we keep it checking for that to ensure we don't accidentally go over.

The values you have here _are_ required if we compile with `--debug`, right? Maybe it makes sense to use those values _only_ for the debug case.

Thoughts?

Reply:
SGTM